Framework/Core/include/Framework/TableTreeHelpers.h

```diff
@@ -122,57 +129,74 @@ class TableToTree
-  TTreeReaderValue<Float_t>* var_f = nullptr;
-  TTreeReaderValue<Double_t>* var_d = nullptr;
-  TTreeReaderValue<UShort_t>* vs = nullptr;
-  TTreeReaderValue<UInt_t>* vi = nullptr;
-  TTreeReaderValue<ULong64_t>* vl = nullptr;
-  TTreeReaderValue<Short_t>* var_s = nullptr;
-  TTreeReaderValue<Int_t>* var_i = nullptr;
-  TTreeReaderValue<Long64_t>* var_l = nullptr;
+  TTreeReaderValue<bool>* mReaderValue_o = nullptr;
+  TTreeReaderValue<uint8_t>* mReaderValue_ub = nullptr;
+  TTreeReaderValue<uint16_t>* mReaderValue_us = nullptr;
+  TTreeReaderValue<uint32_t>* mReaderValue_ui = nullptr;
+  TTreeReaderValue<uint64_t>* mReaderValue_ul = nullptr;
+  TTreeReaderValue<int8_t>* mReaderValue_b = nullptr;
+  TTreeReaderValue<int16_t>* mReaderValue_s = nullptr;
+  TTreeReaderValue<int32_t>* mReaderValue_i = nullptr;
+  TTreeReaderValue<int64_t>* mReaderValue_l = nullptr;
+  TTreeReaderValue<float>* mReaderValue_f = nullptr;
+  TTreeReaderValue<double>* mReaderValue_d = nullptr;
+
+  // all the possible TTreeReaderArray<T> types
+  TTreeReaderArray<bool>* mReaderArray_o = nullptr;
+  TTreeReaderArray<uint8_t>* mReaderArray_ub = nullptr;
+  TTreeReaderArray<uint16_t>* mReaderArray_us = nullptr;
+  TTreeReaderArray<uint32_t>* mReaderArray_ui = nullptr;
+  TTreeReaderArray<uint64_t>* mReaderArray_ul = nullptr;
+  TTreeReaderArray<int8_t>* mReaderArray_b = nullptr;
+  TTreeReaderArray<int16_t>* mReaderArray_s = nullptr;
+  TTreeReaderArray<int32_t>* mReaderArray_i = nullptr;
+  TTreeReaderArray<int64_t>* mReaderArray_l = nullptr;
+  TTreeReaderArray<float>* mReaderArray_f = nullptr;
+  TTreeReaderArray<double>* mReaderArray_d = nullptr;
 
   // all the possible arrow::TBuilder types
-  arrow::BooleanBuilder* bui_o = nullptr;
-  arrow::UInt8Builder* bb = nullptr;
-  arrow::FloatBuilder* bui_f = nullptr;
-  arrow::DoubleBuilder* bui_d = nullptr;
-  arrow::UInt16Builder* bs = nullptr;
-  arrow::UInt32Builder* bi = nullptr;
-  arrow::UInt64Builder* bl = nullptr;
-  arrow::Int16Builder* bui_s = nullptr;
-  arrow::Int32Builder* bui_i = nullptr;
-  arrow::Int64Builder* bui_l = nullptr;
-
-  bool mstatus = false;
-  EDataType marrowType;
-  const char* mcolumnName;
-
-  arrow::MemoryPool* mpool = arrow::default_memory_pool();
-  std::shared_ptr<arrow::Field> mfield;
-  std::shared_ptr<arrow::Array> marray;
+  std::shared_ptr<arrow::FixedSizeListBuilder> mTableBuilder_list;
+
+  arrow::BooleanBuilder* mTableBuilder_o = nullptr;
+  arrow::UInt8Builder* mTableBuilder_ub = nullptr;
+  arrow::UInt16Builder* mTableBuilder_us = nullptr;
+  arrow::UInt32Builder* mTableBuilder_ui = nullptr;
+  arrow::UInt64Builder* mTableBuilder_ul = nullptr;
+  arrow::Int8Builder* mTableBuilder_b = nullptr;
+  arrow::Int16Builder* mTableBuilder_s = nullptr;
+  arrow::Int32Builder* mTableBuilder_i = nullptr;
+  arrow::Int64Builder* mTableBuilder_l = nullptr;
+  arrow::FloatBuilder* mTableBuilder_f = nullptr;
+  arrow::DoubleBuilder* mTableBuilder_d = nullptr;
```

Comment:
Do we really need to have all these? Not for now, but you should try to get rid of them.

Reply:
will need some more thoughts